### PR TITLE
docs:  Add TSF knowledge base page

### LINKED ISSUE #832: docs: Create a text note for TSF and equivalent for linux & macOS
Yep, I think we move the content to a technical note on help.keyman.com and out of the core keyman for windows docs.

_Originally posted by @mcdurdin in https://github.com/keymanapp/keyman/issues/9391#issuecomment-1763197905_

### PR DESCRIPTION
**Pull request by rc-swag**
Fixes #832

### CODE REVIEW

File: knowledge-base/kb0110.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+a much more intuitive manner when using TSF.
+
+## What applications support TSF?
+
+TSF is supported by a range of software and Windows components. 
+This includes: MS Office Applications, Firefox, Adobe Create Cloud Apps and Notepad in Windows 11
+
+## Context aware applications on other platforms
+
+Certain Linux and MacOS applications also have the ability to read the current 'context' from applications.
```
Comment (rc-swag) [inline knowledge-base/kb0110.md:28]: What about iOS In android and iOS are is the ability to read the current context always available?

File: knowledge-base/kb0110.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Text Services Framework and context aware applications
+
+
+## What is the Text Services Framework?
+
+The Text Services Framework, or TSF, is a set of components for modern applications that support advanced input and text processing. It
```
Comment (mcdurdin) [inline knowledge-base/kb0110.md:6]: Can you format this document to our standard line length of 80 chars for .md? I often use Rewrap extension (Alt+Q wraps, Alt+Shift+Q unwraps) on VSCode for this.

File: knowledge-base/kb0110.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+'context' is the characters on the screen around the insertion point.**
+Earlier versions of Keyman remembered the context while inputting text,
+but as soon as an arrow key was pressed, or the mouse clicked, Keyman
+would forget the context. This means that existing text can be edited in
+a much more intuitive manner when using TSF.
+
+## What applications support TSF?
+
+TSF is supported by a range of software and Windows components. 
+This includes: MS Office Applications, Firefox, Adobe Create Cloud Apps and Notepad in Windows 11
```
Comment (mcdurdin) [inline knowledge-base/kb0110.md:24]: The list is a bit longer than this!

```suggestion
For example, this includes: MS Office Applications, Firefox, Adobe Create Cloud Apps, Wordpad, SIL Fieldworks, and Notepad in Windows 11.
```

File: knowledge-base/kb0110.md (hunk 1 above)
Comment (mcdurdin) [reply knowledge-base/kb0110.md:28]: All apps on iOS and Android can read the context. Similarly, on web, context is always available.

**Review by mcdurdin — CHANGES_REQUESTED** [review]

File: knowledge-base/kb0110.md (hunk 1 above)
Comment (rc-swag) [reply knowledge-base/kb0110.md:28]: > All apps on iOS and Android can read the context. Similarly, on web, context is always available.

I thought that maybe the case I am going to add your reply to the document.

File: knowledge-base/kb0110.md (hunk 3 above)
Comment (rc-swag) [reply knowledge-base/kb0110.md:24]: I will remove `this includes`, as for example makes it redundant.

**Review by mcdurdin — APPROVED** [review]

**Comment by rc-swag** [comment]
@darcywong00 or @mcdurdin  This CI / build-test is failing and when I check the link the image complained about is missing. Should I raise an Issue or is this working already over on staging?

**Comment by mcdurdin** [comment]
> This CI / build-test is failing and when I check the link the image complained about is missing. Should I raise an Issue or is this working already over on staging?

Already an issue open at keymanapp/keyboards#2412

**Pull request merged**